File: hw/elbay/u3/hw24CompositePattern/hw24CompositePatternnn/js/model/Directory.js
const FileSystemItem = require("./FileSystemItem");

class Directory extends FileSystemItem {
    constructor(name) {
        super();
        this.name = name;
        this.contents = [];
    }

    describe() {
        super.describe();
        for (const item of this.contents) {
            item.describe();
        }
    }

    add(item) {
        this.contents.push(item);
    }
}

module.exports = Directory;

File: hw/elbay/u3/hw24CompositePattern/hw24CompositePatternnn/java/src/main/java/model/Directory.java
package model;

import java.util.Vector;

public class Directory extends FileSystemItem {
    protected Vector<FileSystemItem> contents = new Vector<>();

    public Directory(String aName) {
        this();
        name = aName;
    }

    public Directory() {
    }

    @Override
    public void describe() {
        super.describe();
        if (contents.size() > 0) {
            for (int i = 0; i < contents.size(); ++i) {
                contents.elementAt(i).describe();
            }
        }
    }

    public void add(FileSystemItem anItem) {
        this.contents.addElement(anItem);
    }
}
